src/change_stream.ts

```diff
@@ -600,8 +600,24 @@ export class ChangeStream<
 
     const cursorOptions: ChangeStreamCursorOptions = filterOptions(options, CURSOR_OPTIONS);
 
+    const client: MongoClient | null =
+      this.type === CHANGE_DOMAIN_TYPES.CLUSTER
+        ? (this.parent as MongoClient)
+        : this.type === CHANGE_DOMAIN_TYPES.DATABASE
+        ? (this.parent as Db).s.client
+        : this.type === CHANGE_DOMAIN_TYPES.COLLECTION
+        ? (this.parent as Collection).s.db.s.client
+        : null;
+
+    if (client == null) {
```

Comment:
question(s): even though this error should never be thrown, in theory _createChangeStreamCursor now throws.  
1. Should we add test cases for this scenario?
2. Should we update the usages of this method to ensure that we're catching the error and handling it appropriately in callback code?  Specifically, I'm looking at `_processError` in change_stream.ts.

Reply:
We could add a test, the thing is this is unreachable because the constructor will throw if one of these three symbols isn't used. This is one of those fatal errors where something quite unexpected has occurred. The test would be making a change stream and then reaching in an changing the type? I can do that, would that make sense?

Reply:
Yeah, that's a fair point.  I'll leave this thread open for additional reviews for context.